### ISSUE TITLE
Method before does not exist.

### ISSUE THREAD

**Issue by thenickcox** [andersao/laravel-request-logger#12]
I've tried installing this both ways suggested in the readme. 

Here's the output from adding it to the `composer.json` file:

```
$ composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing paragonie/random_compat (v1.3.0)
  - Installing paragonie/random_compat (v1.4.1)
    Downloading: 100%

  - Removing laravel/framework (v5.2.23)
  - Installing laravel/framework (v5.2.24)
    Downloading: 100%

Writing lock file
Generating autoload files
> php artisan clear-compiled


  [BadMethodCallException]
  Method before does not exist.


Script php artisan clear-compiled handling the post-update-cmd event returned with an error


  [RuntimeException]
  Error Output:


update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...
```

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by emahuni** [comment]
Hi. 

Same here and I am using ver 5.2.24. It seems like my app is now doing that for everything i try to run through artisan.

I am new to Laravel and I am actually learning things. I love this framework. 

What i however noticed from the documentation that is out of place is this:

'Prettus\RequestLogger\Providers\LoggerServiceProvider',

and i thought well, that's kinda strange; mostly software follows a certain pattern on data. So noticed things look the same and then i ended up changing it to this:

Prettus\RequestLogger\Providers\LoggerServiceProvider::class,

Judging from that i think this was written for a previous version of Laravel that implemented things a bit differently.

Looked around the code a bit and noticed other packages with that before method. Could it be there is a later version other than 1.0.* (1 in this case) that we are supposed to be requiring through composer and that the readme.md file is not updated to yet?

Solution:

I had to remove composer require entry and remove the above line from config/app.php file, ran composer update to return everything to normal.

**Comment by markdegrootnl** [comment]
It seems to be related to the Before() and After() methods being deprecated and now removed from Laravel 5.2. See: #10

**Comment by codetroll** [comment]
Any news on this?

**Comment by bandtank** [comment]
+1 still broken...

**Comment by peterchaula** [comment]
Also broken here.

**Comment by riazXrazor** [comment]
its still broken.. :(
anyone found a fix ?

**Comment by atomkirk** [comment]
This PR fixes it: https://github.com/andersao/laravel-request-logger/pull/13

Although it looks like it removes benchmarking without replacing it, so I'm not sure that's ideal.

**Comment by itsazzad** [comment]
Again! with `Laravel 5.3`
```
[BadMethodCallException]       
  Method before does not exist.  
```

**Comment by ramyfarid922** [comment]
I am using Laravel 5.2 and as soon as I tried publishing the package via artisan I got this error

**Comment by itsazzad** [comment]
I have made a fork with few updates to match with L5.3 
@ramyfarid922 you may check.
https://github.com/itsazzad/laravel-request-logger

**Comment by Ra1da35ma** [comment]
so this is still why I wunt use this package

**Comment by dev-mohammed** [comment]
it's still broken ...

**Comment by sergeimishurin** [comment]
Monthly reminder: still broken.

**Comment by andersao** [comment]
A new version was released [1.2.0](https://github.com/andersao/laravel-request-logger/releases/tag/1.2.0). Check it out.
